File: features/notes/src/main/java/com/nivasafinance/features/notes/repository/NotesRepository.java
package com.nivasafinance.features.notes.repository;

import com.nivasafinance.features.notes.entity.Notes;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import java.util.Optional;
import java.util.UUID;

@Repository
public interface NotesRepository extends JpaRepository<Notes, Long> {
    Optional<Notes> findByIdentifier(UUID identifier);
}

File: features/notes/src/main/java/com/nivasafinance/features/notes/repository/NotesRepository.kt
package com.nivasafinance.features.notes.repository

import com.nivasafinance.features.notes.entity.Notes
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface NotesRepository : JpaRepository<Notes, UUID>
